src/utils.ts

```diff
@@ -76,28 +72,45 @@ export class Utils {
         }
     }
 
+    /**
+     * Returns true if OpenID Connect authentication should be used.
+     * @param jfrogCredentials - Credentials retrieved from the environment variables
+     * @returns true if OpenID Connect authentication should be used
+     */
+    private static shouldUseOpenIDConnect(jfrogCredentials: JfrogCredentials): boolean {
+        if (!jfrogCredentials.jfrogUrl) {
+            // If no JFrog URL is specified, we can't use OpenID Connect
+            return false;
+        }
+        if (jfrogCredentials.password || jfrogCredentials.accessToken) {
+            // If credentials are specified - use them instead
+            return false;
+        }
+        if (!process.env.ACTIONS_ID_TOKEN_REQUEST_URL) {
+            // Only if the user added the 'id-token: write' permission, this environment variable was set
```

Comment:
```suggestion
            // To use OpenIDConnect auth the user should set the ‘id-token: write’ permission that sets the process.env.ACTIONS_ID_TOKEN_REQUEST_URL. If it's empty then shouldn't use Oidc.
```

also make this the first 'if'